src/common/status.h

```diff
@@ -397,3 +397,16 @@ bool StatusIsOK(const T& v) {
     if (!StatusIsOK(status)) return std::forward<decltype(status)>(status); \
     std::forward<decltype(status)>(status);                                 \
   }))
+
+// NOLINTNEXTLINE
+#define RETURN_IF_ERROR(...)                                                \
+  ({                                                                        \
+    auto&& status = (__VA_ARGS__);                                          \
+    if (!StatusIsOK(status)) return std::forward<decltype(status)>(status); \
+  })
+
+// NOLINTNEXTLINE
+#define RETURN_IF_ERR_FROM_ROCKSDB(s, code, msg) \
```

Comment:
As I already said, we only need a function/method/ctor, instead of a macro. If we can complete some task via function/method, please never introduce a new macro.

Reply:
> As I already said, we only need a function/method/ctor/operator T, instead of a macro. If we can complete some task via function/method, please never introduce a new macro.

Regarding the comment, after careful consideration, I realize that I may not have fully grasped your intention. Specifically, I feel that without introducing a new macro, it becomes challenging to directly return a new Status object.